MVP/python/CouchHeartbeat.py

```diff
@@ -0,0 +1,70 @@
+"""
+ CouchDB heartbeat
+ Check database and if not running, restart
+ Restarting CouchDB does not always go clear, reboot is better
+"""
+
+import os
+import requests
+from LogUtil import get_logger
+
+class CouchHeartbeat(object):
+    """Heartbeat object    """
+
+    logger = None
+
+    def __init__(self):
+        """Standard constructor
+        Get and hold a Python logger
+        """
+        self.logger = get_logger('CouchHeartbeat')
+
+    def check(self, port):
+        """Ping the database
+        Should return the welcome message
+        Throws an exception if cannot make a connection to the database
+
+        Args:
+            port: port the database is communicating on
+        Returns:
+            None
+        Raises:
+            None
+        """
+        try:
+            request = requests.get('http://localhost:' + port)
+            if request.json()["couchdb"] == 'Welcome':
+                self.logger.info('Port: %s Couch Up' % (port))
+        except requests.ConnectionError as error:
+            self.logger.warning('Port: %s Couch Down %s' % (port, error))
+            restart()
+
+    def restart(self):
+        """System restart (reboot)
+        Args:
+            None
+        Returns:
+            None
+        Raises:
+            None
+        """
+        cmd = 'sudo reboot'
+        self.logger.warning('System restart: %s' % (cmd))
+        os.system(cmd)
+
+def test():
```

Comment:
Tests get embedded directly in the file? huh. interesting

Reply:
If I was going full blown and using Pytest, these would all be separate files, but since this is light-weight testing it is easiest to have the test in with the rest of the code (I run the test as I am developing the code), and cuts down on the directory clutter.
Where I had the test as a class method, I am pulling it out and making it a function outside the class, that way if we ever move to Pytest, it is already an independent function.  The down-side of putting the functions in the code file is that is uses a bit more memory, as it is (possibly) loaded on import.
This just happens to by my style, I have not seen any best practices around test functions.  If you have some other suggestions, I am open to them.